### DISCUSSION TITLE
FTS5 - BM25 expexted argument count

### DISCUSSION THREAD

**Discussion by RedyAu** [simolus3/drift#3292] [Q&A] [answered]
Hello!
I have a query defined in a `.drift` file like this:
```sql
song_fulltext_search(:match_string AS TEXT):
  SELECT
      CAST(BM25(songs_fts, 0.0, 0.0, 10.0, 0.5, 5.0, 5.0, 2.0, 0.0) AS REAL) AS rank
      ,uuid
      ,source_bank_id
      ,pitch_field
      ,CAST(SNIPPET(songs_fts, 2, '<?', '?>', '...', 30) AS TEXT) AS match_title
      ,CAST(SNIPPET(songs_fts, 3, '<?', '?>', '...', 30) AS TEXT) AS match_lyrics
      ,CAST(SNIPPET(songs_fts, 4, '<?', '?>', '...', 30) AS TEXT) AS match_composer
      ,CAST(SNIPPET(songs_fts, 5, '<?', '?>', '...', 30) AS TEXT) AS match_poet
      ,CAST(SNIPPET(songs_fts, 6, '<?', '?>', '...', 30) AS TEXT) AS match_translator
    FROM songs_fts
    WHERE songs_fts MATCH :match_string
    ORDER BY rank;
```
With FTS5, BM25 can be used to assign weights to each column for ranking. A query like above runs fine with SQLite.
However, Drift returns this warning during build:

```log
[WARNING] drift_dev on lib/data/database.dart:
line 6, column 12: BM25 expects 1 arguments, got 9.
  ╷
6 │       CAST(BM25(songs_fts, 0.0, 0.0, 10.0, 0.5, 5.0, 5.0, 2.0, 0.0) AS REAL) AS rank
  │            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ╵
```

Am I doing something wrong, or should I just disable this warning?

**Comment by simolus3** [comment, marked as answer]
Drift is doing something wrong here, I was not aware of the option to assign weights. You can ignore the warning for now, I don't think we have a way to explicitly ignore warnings at the moment.

Edit: Fixed in f692bb1bbbae8add399c5f6c953c619ccb87bcf5.

**Reply by RedyAu** [reply]
Great, thank you!

**Reply by RedyAu** [reply]
Now that we're here, you could consider setting the type of `SNIPPET()` and `HIGHLIGHT()` to `TEXT` and `BM25()` to `REAL`, as the `CAST()`s are only there to help Drift. ;)

**Reply by simolus3** [reply]
This is awkward, but try using `bm25` and `snippet` in lower case, the resolver should recognize the functions and their types in that case... I'll fix this as well.
